specifyweb/frontend/js_src/lib/components/FormPlugins/DateInput.tsx

```diff
@@ -88,6 +90,9 @@ export function DateInput({
   return (
     <>
       <Input.Generic
+        className={
+          !dateSupported && canChangePrecision ? '' : '!w-[unset] grow'
+        }
```

Comment:
While this fix seems to work, I would encourage us not to introduce back complexity to PartialDateUi as it's already quite complex (and any complexity would lead to more bugs in the future)

instead, here is the solution that would involve simplifying it a bit:

take a look at this line:

https://github.com/specify/specify7/blob/d11322782d29b472296d3ac05b780820bdb41cde/specifyweb/frontend/js_src/lib/components/FormPlugins/PartialDateUi.tsx#L52

when date precision field is not visible, there is no need for `grid grid-cols-[fit-content(40%)_1fr] gap-1` - so instead of adding this condition here, with extra classes and `!`, could you just remove the grid related classes from the container?